### PR TITLE
optimize substr performance and fix ASAN global buffer overflow 

### PR DESCRIPTION
**Pull request by xiaokang**
# Proposed changes

fix https://github.com/apache/doris/pull/10169

### CODE REVIEW

File: be/src/vec/common/string_searcher.h [hunk 1]
```diff
@@ -0,0 +1,860 @@
...
+        return ((page_size - 1) & reinterpret_cast<std::uintptr_t>(ptr)) <= page_size - n;
+    }
+#endif
+};
+
+/// Performs case-sensitive and case-insensitive search of UTF-8 strings
+template <bool CaseSensitive, bool ASCII>
+class StringSearcher;
+
+// comment out since it's not used in doris and UTF8 dependency is not easy to meet
```
Comment (HappenLee) [inline be/src/vec/common/string_searcher.h:69]: we should delete unless comment code， same of other code

Comment (xiaokang) [reply be/src/vec/common/string_searcher.h:69]: ok. 

btw, can you have a look at the commit to fix global buffer overflow? It's relevant to vectorization. I think you are the one who is very familiar with vectorization code.

**Review by kpfly — DISMISSED** [review]

**Comment by xiaokang** [comment]
@HappenLee @kpfly Is there any more comments before it can be merged?

**Review by kpfly — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**